### ISSUE TITLE
Add notebook comparing SC weight parameterizations (Dirichlet vs softmax-Normal)

### ISSUE THREAD

**Issue by drbenvincent** [pymc-labs/CausalPy#829]
## Summary

Create a new example notebook that compares the two weight parameterizations available for Bayesian synthetic control: `WeightedSumFitter` (Dirichlet prior) and `SoftmaxWeightedSumFitter` (softmax-over-Normal-logits). The notebook should help users understand when to choose each approach and build intuition about how the prior choice affects the weights and the counterfactual estimate.

## Motivation

After #822 merges, CausalPy will offer two model classes for Bayesian SC that both produce simplex-valued weights (positive, sum to 1) but encode very different prior beliefs and regularization behaviour:

- **Dirichlet** (`WeightedSumFitter`): concentration parameter `a` controls sparsity vs uniformity. With `a=1` the prior is uniform on the simplex. Setting `a < 1` encourages sparsity (weights concentrating on fewer donors), while `a > 1` encourages uniformity. This is the existing default in CausalPy.
- **Softmax-Normal** (`SoftmaxWeightedSumFitter`): places Normal priors on unconstrained logits and maps them to the simplex via softmax. The first logit is pinned to zero to remove the softmax's shift invariance. The prior scale `sigma` on the logits continuously interpolates between DiD-like uniform weights (small sigma → logits shrunk toward zero → all weights ≈ 1/N) and SC-like sparse weights (large sigma → data can push most weight onto a few donors).

The class docstrings in `pymc_models.py` explain this distinction, but a notebook that *shows* it on real data — with posterior weight distributions, counterfactual trajectories, and the effect of varying sigma — would be far more useful for practitioners. Without this notebook, users who see both model classes in the API have no practical guidance on when to use which or what difference it makes.

### Connection to SDiD

The softmax-Normal parameterization was introduced to support Bayesian Synthetic Difference-in-Differences (SDiD, see issue #47 and PR #823). In the frequentist SDiD of Arkhangelsky et al. (2021), the ℓ₂ regularization parameter ζ plays exactly this role — it controls whether the unit weights look more like DiD (uniform) or SC (sparse). The prior scale `sigma` in `SoftmaxWeightedSumFitter` is the Bayesian analogue of 1/ζ. This notebook doesn't need to cover SDiD itself, but it should mention this connection so users understand the broader context.

## Dependencies

- **Requires PR #822 to be merged first.** The `SoftmaxWeightedSumFitter` class does not exist on `main` yet.
- No dependency on PR #823 (Bayesian SDiD). This notebook is purely about SC weight priors.

## Suggested location

`docs/source/notebooks/sc_pymc_weight_comparison.ipynb`, listed under the existing "Synthetic Control" section in `docs/source/notebooks/index.md` (alongside `sc_skl.ipynb`, `sc_pymc.ipynb`, `sc_pymc_brexit.ipynb`).

### 1. Introduction

Brief explanation of the two parameterizations and why the choice of weight prior matters for synthetic control. Cross-reference the existing `sc_pymc.ipynb` workflow for the full SC pipeline — this notebook is a focused companion, not a replacement.

Key points to cover:
- Both models produce simplex weights (positive, sum to 1) — neither allows negative weights or exact zeros.
- They differ in *how* they get to the simplex and what prior beliefs they encode.
- The Dirichlet directly samples on the simplex. The softmax-Normal samples in unconstrained space and maps to the simplex via softmax, which tends to give better HMC/NUTS gradient geometry.

### 2. Setup

Load the California Proposition 99 dataset (available as `cp.load_data("california_prop99")` after #823, or use the built-in `sc` dataset which is already available). Define control/treated units and treatment time.

### 3. Fit both models on the same data

Run `cp.SyntheticControl` twice on the same dataset — once with `WeightedSumFitter` (the default) and once with `SoftmaxWeightedSumFitter`. Use identical `sample_kwargs` so the only difference is the weight prior.

```python
result_dirichlet = cp.SyntheticControl(
    df, treatment_time,
    control_units=control_units,
    treated_units=[treated_unit],
    model=cp.pymc_models.WeightedSumFitter(sample_kwargs=sample_kwargs),
)

result_softmax = cp.SyntheticControl(
    df, treatment_time,
    control_units=control_units,
    treated_units=[treated_unit],
    model=cp.pymc_models.SoftmaxWeightedSumFitter(sample_kwargs=sample_kwargs),
)
```

### 4. Compare posterior weights

Side-by-side bar charts of the posterior median weights with 94% credible intervals. Highlight how the Dirichlet prior distributes weight vs how the softmax-Normal prior does. Key observations to draw out:
- Which control units receive the most weight under each parameterization?
- Are the credible intervals wider or narrower?
- Do both models agree on which donors are most important?

### 5. Compare counterfactual trajectories

Overlay the two synthetic controls on the observed treated unit's data. Show that both recover similar counterfactuals but potentially with different uncertainty profiles. Compare the causal effect estimates (point estimate and HDI).

### 6. Regularization path plot (the sigma knob)

**This is the centrepiece of the notebook.** Re-fit `SoftmaxWeightedSumFitter` at several values of sigma (e.g. 6–8 values spanning 0.01 to 10 on a log scale) and produce a **regularization path plot**:

- **X-axis:** sigma (log scale), the prior scale on the logits
- **Y-axis:** posterior median weight for each control unit
- **Each line** represents one control unit's weight as sigma varies
- Optionally, add thin credible bands (ribbons) around each line to show posterior uncertainty

This is directly analogous to the coefficient path plots from ridge/LASSO regression that many practitioners will recognise from textbooks or sklearn tutorials. In the ridge/LASSO world, the x-axis is the regularization penalty λ and you watch coefficients shrink toward zero as λ increases. Here, you watch the weights transition from uniform to sparse as sigma increases.

The visual story should be immediately clear:
- At `sigma ≈ 0.01`: all lines are flat at ~1/N (uniform weights, DiD-like). The prior forces all logits close to zero, so softmax produces near-equal weights.
- At `sigma ≈ 1.0` (the default): lines begin to diverge — some donors receive more weight, others less. The data has enough freedom to differentiate donors, but with moderate regularization.
- At `sigma ≈ 10.0`: one or two lines dominate, the rest collapse near zero (sparse weights, SC-like). The prior is essentially flat, so the data drives the weights toward a sparse solution.

Add a **vertical reference line at sigma=1.0** to show that the default sits in a sensible middle ground.

**The Bayesian twist:** Unlike the frequentist ridge path where each point is a single coefficient value, here each "line" can show a posterior credible band (thin ribbon). This reveals that at small sigma the posterior is very tight (weights are forced to be similar, so there's little uncertainty) while at large sigma the posterior is wider (the data gets more say but there's less regularization to constrain the solution). This is a distinctly Bayesian insight you don't get from the classical regularization path.

**Practical note:** generating this plot requires fitting the model at several sigma values. Each fit is a full MCMC run, so use lightweight `sample_kwargs` (e.g. `tune=500, draws=500, chains=2`) and the small built-in `sc` dataset to keep the notebook's CI build time reasonable.

To override sigma, pass a custom prior:

```python
from pymc_extras.prior import Prior

model = cp.pymc_models.SoftmaxWeightedSumFitter(
    sample_kwargs=sample_kwargs,
    priors={
        "beta_raw": Prior(
            "Normal", mu=0, sigma=sigma_value,
            dims=["treated_units", "coeffs_raw"],
        )
    },
)
```

### 7. Practical guidance

Wrap up with a summary of when to prefer each approach:

- **Dirichlet (`WeightedSumFitter`):** familiar, well-understood, good default for standard SC. The concentration parameter `a` has an intuitive "pseudo-count" interpretation. Use this when you have no reason to think about the DiD/SC spectrum and want a straightforward prior.
- **Softmax-Normal (`SoftmaxWeightedSumFitter`):** choose this when you want smooth, continuous control over the DiD-to-SC spectrum via sigma. Also the natural choice when building toward SDiD (where sigma plays the role of the ζ regularization parameter from Arkhangelsky et al., 2021). The softmax parameterization tends to have better NUTS sampling geometry because the sampler operates in unconstrained space.

Mention that both models are drop-in replacements for each other in `cp.SyntheticControl` — no other code changes are needed.

## Notebook conventions

- Follow CausalPy notebook conventions: MyST directives for callouts (e.g. `:::{note}`), glossary linking on first mention (e.g. `{term}\`synthetic control\``), bibliography section at the bottom using `:::{bibliography}` with `:filter: docname in docnames`.
- Use `sample_kwargs` that keep the notebook fast to build in CI (e.g. `target_accept=0.95, random_seed=42`). The regularization path section will need multiple fits, so keep the per-fit budget lean.
- Add relevant references to `docs/source/references.bib` if not already present (Arkhangelsky et al. 2021 is already there after #823; if not yet merged, add it).
- The notebook should be relatively concise — this is a focused comparison, not a full workflow. The existing `sc_pymc.ipynb` covers the complete SC pipeline and should be cross-referenced, not duplicated.

---

cc @thomaspinder — tagging you since this builds directly on your work in #822 and your [blog post on Bayesian SDiD](https://thomas-pinder.com/writing/bayesian_sdid/). Feel free to pick this up if you're interested, but no obligation.
